### ISSUE TITLE
The program crashed when clicking a button

### ISSUE THREAD

**Issue by borjafg** [pubpub-zz/ppInk#158]
Program configuration file:   [config.ini.zip](https://github.com/pubpub-zz/ppInk/files/7552686/config.ini.zip)

Content of "crash.txt":

```
UIThreadException

version 1.8.1.24635 built on 28/08/2021 13:41
Oops, ppInk crashed! Please include the following information if you plan to contact the developers (a copy of the following information is stored in crash.txt in the application folder):

Índice fuera de los límites de la matriz.

Stack Trace:
   en gInk.FormCollection.CustomizeAndOpenSubTools(Int32 active, String title, String[] icons, String TextHintsStr, Func`2[] clickFuncts) en D:\gink.git\ppInk\src\FormCollection.cs:línea 6505
   en gInk.FormCollection.btTool_Click(Object sender, EventArgs e) en D:\gink.git\ppInk\src\FormCollection.cs:línea 6122
   en System.Windows.Forms.Control.OnClick(EventArgs e)
   en System.Windows.Forms.Button.OnClick(EventArgs e)
   en System.Windows.Forms.Button.OnMouseUp(MouseEventArgs mevent)
   en System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   en System.Windows.Forms.Control.WndProc(Message& m)
   en System.Windows.Forms.ButtonBase.WndProc(Message& m)
   en System.Windows.Forms.Button.WndProc(Message& m)
   en System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

**Comment by borjafg** [comment]
The error occurs when I click one of these buttons.

![image](https://user-images.githubusercontent.com/6740268/142208795-771ea7ba-fbe5-47e1-a4d7-434bffb86fa1.png)

**Comment by pubpub-zz** [comment]
Hi @borjafg 
sorry for late answer. at first sight it looks like it is an issue with translations similar to #152 :
Can you confirm that you have "enabled secondary toolbar"
if so, you should update the language translation file you are using to have as many hints as number of icons for all entries that are ending with "SubToolsHints"
the best solution I can propose you is to copy and translate the entries from en-us.txt

**Comment by pubpub-zz** [comment]
@borjafg 
any feedbacks ?

**Comment by borjafg** [comment]
Sorry. I was busy last week, but I think I can try the change you suggest this week.

**Comment by pubpub-zz** [comment]
@borjafg 
Did it work ?

**Comment by borjafg** [comment]
I was downloading a program to edit the properties in parallel and see the differences.

I'm going to do some tests and I'll inform you of the results.

-------


It's this configuration that produces the error.

![image](https://user-images.githubusercontent.com/6740268/144711743-cff53564-30f5-4a53-9185-a395e37cc937.png)

**Comment by borjafg** [comment]
Ok. These entries are missing in the es-es file


```
ArrowDlg
ButtonAddText
ButtonDelText
ButtonExitText
ButtonNameScaleRotate
ButtonNextText
ButtonPrevText
ButtonSaveText
KeyNamesStr
OptionsClick4Stroke
OptionsSpotLightBox
OptionsSpotLightRadius
OptionsSpotLightOnAlt
PatternStoreParamTxt
ScaleSubToolsHints
```


I can add the translation and make a pull request, but I have one question about the entry "KeyNamesStr". Should its content be translated or should it be the same in all message files?

![image](https://user-images.githubusercontent.com/6740268/144714067-a478f9c1-3666-4d9f-9a07-96a4740af5cd.png)

**Comment by pubpub-zz** [comment]
es-es is obsolete. now spanish translation in available in es.txt which was buggy also 
Can you try with this translation ?
[es.txt](https://github.com/pubpub-zz/ppInk/files/7654214/es.txt)

**Comment by borjafg** [comment]
These are the differences I found. Unfortunately the problem persists after modifying the properties.

---
---


**ButtonNameHand**

![image](https://user-images.githubusercontent.com/6740268/144717907-96331935-adc9-4d88-826a-ffd344cc264f.png)


---


**ButtonNameLine**

![image](https://user-images.githubusercontent.com/6740268/144717856-7a3ee9c4-e579-4187-b0c8-93e6ddc9035e.png)


---


**ButtonNameOval**


![image](https://user-images.githubusercontent.com/6740268/144717805-31812fef-d33d-44da-99dd-4986baff08f7.png)


---


**ButtonNamePan**

![image](https://user-images.githubusercontent.com/6740268/144717764-07b0c5de-9c16-4d49-9b9b-a7d6402ec77d.png)


---


**ButtonNameRect**

![image](https://user-images.githubusercontent.com/6740268/144717716-2fde42d0-5ad6-4541-9c3f-82271085d17a.png)


---


**HandSubToolsHints**

![image](https://user-images.githubusercontent.com/6740268/144717678-69b3f3ed-058b-427e-9041-cfc4ab4cb391.png)

----


**KeyNamesStr**

![image](https://user-images.githubusercontent.com/6740268/144717568-b02298a5-eaa0-4dc6-85a9-e7a8fd0244c8.png)


---


**LineSubToolsHints**

![image](https://user-images.githubusercontent.com/6740268/144717302-97756f29-a220-4d22-87c0-cf8cf446b957.png)


---


**ListFillingsText**

![image](https://user-images.githubusercontent.com/6740268/144717263-57a3e3e1-24af-4ee8-9349-e88330b7968d.png)


---


**OvalSubToolsHints**

![image](https://user-images.githubusercontent.com/6740268/144716971-fb1b860c-8fc7-411b-98c0-d2378b38c544.png)


---


**RectSubToolsHints**

![image](https://user-images.githubusercontent.com/6740268/144716921-ca8eaaa2-a572-4016-ab36-2a1eb8ff5698.png)

**Comment by pubpub-zz** [comment]
You are right. Can you update the file and send it ?

**Comment by borjafg** [comment]
This is the updated file:   [es.txt](https://github.com/pubpub-zz/ppInk/files/7654447/es.txt)

**Comment by pubpub-zz** [comment]
@borjafg,
I will close this thread as a duplicate of #152 and as you've kindly provided the fix. Thanks.
Do not hesitate to star the project if you enjoy it 😊
